File: src/test/java/de/roamingthings/workbench/testcontainers/GreetingPage.java
package de.roamingthings.workbench.testcontainers;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class GreetingPage extends PageObject {

    @FindBy(id = "name")
    private WebElement nameField;

    public GreetingPage(WebDriver driver) {
        super(driver);
    }

    public boolean isInitialized() {
        return nameField.isDisplayed();
    }

    public String name() {
        return nameField.getText();
    }

}

File: src/test/kotlin/de/roamingthings/workbench/testcontainers/kt/GreetingPage.kt
package de.roamingthings.workbench.testcontainers.kt

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.FindBy

class GreetingPage(driver: WebDriver) : PageObject(driver) {

    @FindBy(id = "name")
    private val nameField: WebElement? = null

    val isInitialized: Boolean
        get() = nameField!!.isDisplayed

    fun name(): String {
        return nameField!!.text
    }

}
